frontend/src/features/admin-form/create/payment/PaymentDrawer.tsx

```diff
@@ -329,12 +316,28 @@ export const PaymentDrawer = ({
     return resetData
   }, [paymentsField, resetData, setData])
 
+  const isPaymentEligible = isEncryptMode && isStripeConnected
+
   // Allows for payment data refresh in encrypt mode
-  if (!paymentData && isEncryptMode) return null
+  if (!paymentData && isPaymentEligible) return null
+
+  const isPaymentDisabled = !isPaymentEligible
```

Comment:
nit: should we instead define it by presence of `paymentDisabledMessage`? This will allow us to take advantage of the single source of truth of logic.

```typescript
const paymentDisabledMessage = // ... logic ...

const isPaymentDisabled = !!paymentDisabledMessage

// Allows for payment data refresh in encrypt mode
if (!paymentData && !isPaymentDisabled) return null

return (
  <CreatePageDrawerContainer>
    {paymentDisabledMessage && (...)}
    // ...
)